edfi_api_client/session.py

```diff
@@ -106,6 +123,69 @@ def authenticate(self) -> dict:
             else:
                 return self.auth_headers
 
+        if self.use_persisted_token:
+            # check to see if on-disk version has been updated
```

Comment:
Let's move the logic in this conditional into its own helper method that returns `auth_payload`.